warp10/src/main/java/io/warp10/script/functions/GET.java

```diff
@@ -17,74 +17,93 @@
+      }
+    } catch (WarpScriptException | IndexOutOfBoundsException e) {
+      throw new WarpScriptException(getName() + " failed.", e);
+    }
 
-      if (coll instanceof List) {
-        int size = ((List) coll).size();
+    stack.push(value);
 
-        idx = computeAndCheckIndex(idx, size);
+    return stack;
+  }
 
-        value = ((List) coll).get(idx);
+  /**
+   * Get a value from a List, Map, byte[] or String using respectively a Long, Object, Long or Long key.
+   * @param key Either on Object for a Map or a Long for List, byte[] or String. Negative indexing is possible, in that case the corresponding index is size + index.
+   * @param collection Either a List, Map, byte[] or String instance.
+   * @return The value at the given key.
+   * @throws WarpScriptException If the collection type cannot be handled or the key is invalid for the collection type.
+   */
+  public static Object get(Object key, Object collection) throws WarpScriptException {
+    if (collection instanceof List) {
+      if (!(key instanceof Long)) {
+        throw new WarpScriptException("Getting on List requires a Long.");
+      }
 
-      } else {
-        int size = ((byte[]) coll).length;
+      int idx = ((Long) key).intValue();
+      int size = ((List) collection).size();
+      idx = computeAndCheckIndex(idx, size);
+      return ((List) collection).get(idx);
 
-        idx = computeAndCheckIndex(idx, size);
+    } else if (collection instanceof Map) {
+      return ((Map) collection).get(key);
 
-        value = (long) (((byte[]) coll)[idx] & 0xFFL);
+    } else if (collection instanceof String) {
+      if (!(key instanceof Long)) {
+        throw new WarpScriptException("Getting on String requires a Long.");
       }
 
-    } else if (coll instanceof byte[]) {
-      throw new WarpScriptException(getName() + " expects the key to be an integer when operating on a byte array.");
-
-    } else if (!(key instanceof List)) {
-      throw new WarpScriptException(getName() + " expects the key to be an integer or a list of integers when operating on a List.");
+      int idx = ((Long) key).intValue();
+      int size = ((String) collection).length();
+      idx = computeAndCheckIndex(idx, size);
+      return String.valueOf(((String) collection).charAt(idx));
 
-    } else {
-      for (Object o: (List) key) {
-        if (!(o instanceof Long)) {
-          throw new WarpScriptException(getName() + " expects the key to be an integer or a list of integers when operating on a List.");
-        }
+    } else if (collection instanceof byte[]) {
+      if (!(key instanceof Long)) {
+        throw new WarpScriptException("Getting on bytes requires a Long.");
       }
 
-      value = nestedGet((List) coll, (List<Long>) key);
-    }
-    
-    stack.push(value);
+      int idx = ((Long) key).intValue();
+      int size = ((byte[]) collection).length;
+      idx = computeAndCheckIndex(idx, size);
+      return ((byte[]) collection)[idx] & 0xFFL;
```

Comment:
It would probably be better to return a byte array of length 1 rather than a LONG, it would be more coherent with the way GET behaves for STRINGs.

Reply:
That would change the current behavior.